internal/jobcontainers/logon.go

```diff
@@ -0,0 +1,67 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// Takes in a DOMAIN\Username or just Username and will return a token
+// for the account if successful.
+func processToken(user string) (windows.Token, error) {
+	var (
+		domain   string
+		userName string
+		token    windows.Token
+	)
+
+	split := strings.Split(user, "\\")
+	if len(split) == 2 {
+		domain = split[0]
+		userName = split[1]
+	} else if len(split) == 1 {
+		userName = split[0]
+	} else {
+		return token, fmt.Errorf("invalid user string `%s`", user)
+	}
+
+	// If empty, ContainerUser or ContainerAdministrator just let it inherit the token
+	// from whatever is used to launch it (containerd-shim etc). Regular container images
+	// are usable for job containers so we need to handle the ContainerUser and ContainerAdministrator
+	// cases.
+	if user == "" || user == "ContainerUser" || user == "ContainerAdministrator" {
+		return openCurrentProcessToken()
+	}
+
+	var logonType uint32
+	// User asking to run as a local system account (NETWORK SERVICE, LOCAL SERVICE, SYSTEM)
```

Comment:
Nit: can you have this inside the if? Just makes it clearer since this really only pertains to the first statement and the else has it's own comment

Reply:
Sure!